theano/scalar/basic.py

```diff
@@ -2166,6 +2168,14 @@ def c_code(self, node, name, (x,), (z,), sub):
         return "%(z)s = -%(x)s;" % locals()
 neg = Neg(same_out, name='neg')
 
+pprint.assign(add, printing.OperatorPrinter('+', -2, 'either'))
+pprint.assign(mul, printing.OperatorPrinter('*', -1, 'either'))
+pprint.assign(sub, printing.OperatorPrinter('-', -2, 'left'))
+pprint.assign(neg, printing.OperatorPrinter('-', 0, 'either'))
+pprint.assign(true_div, printing.OperatorPrinter('/', -1, 'left'))
+pprint.assign(int_div, printing.OperatorPrinter('//', -1, 'left'))
+pprint.assign(pow, printing.OperatorPrinter('**', 1, 'right'))
+pprint.assign(mod, printing.OperatorPrinter('%', -1, 'lef'))
```

Comment:
This raises the issue that OperatorPrinter doesn't check for valid kinds of
associativity.

@mohammadpz While you're fixing the typo can you add an assertion here
https://github.com/Theano/Theano/blob/master/theano/printing.py#L277 that
checks assoc is either "left", "right" or "either"? Good idea to define
VALID_ASSOC = set(['left', 'right', 'either']) at the class level and then
check that "assoc in VALID_ASSOC". This assertion would've caught the typo.

On Wed, Jan 21, 2015 at 2:51 PM, Mohammad Pezeshki <notifications@github.com
> wrote:

> In theano/scalar/basic.py
> <https://github.com/Theano/Theano/pull/2407#discussion_r23328524>:
>
> > @@ -2166,6 +2168,14 @@ def c_code(self, node, name, (x,), (z,), sub):
> >          return "%(z)s = -%(x)s;" % locals()
> >  neg = Neg(same_out, name='neg')
> >
> > +pprint.assign(add, printing.OperatorPrinter('+', -2, 'either'))
> > +pprint.assign(mul, printing.OperatorPrinter('*', -1, 'either'))
> > +pprint.assign(sub, printing.OperatorPrinter('-', -2, 'left'))
> > +pprint.assign(neg, printing.OperatorPrinter('-', 0, 'either'))
> > +pprint.assign(true_div, printing.OperatorPrinter('/', -1, 'left'))
> > +pprint.assign(int_div, printing.OperatorPrinter('//', -1, 'left'))
> > +pprint.assign(pow, printing.OperatorPrinter('**', 1, 'right'))
> > +pprint.assign(mod, printing.OperatorPrinter('%', -1, 'lef'))
>
> @lamblin <https://github.com/lamblin> , @dwf <https://github.com/dwf> ,
> Sorry for typo. I'll fix it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/2407/files#r23328524>.
>